internal/credentials/kubernetes/ecr/access_key.go

```diff
@@ -57,11 +57,16 @@ func (a *accessKeyCredentialHelper) getCredentials(
 	repoURL string,
 	secret *corev1.Secret,
 ) (*credentials.Credentials, error) {
-	if credType != credentials.TypeImage || secret == nil {
+	if credType == credentials.TypeGit || secret == nil {
```

Comment:
```suggestion
	if (credType != credentials.TypeImage && credType != credentials.TypeHelm &&) || secret == nil {
```

This is safer when we add more credential types -- which we inevitably will.